src/sot_gepetto_viewer/graph.py

```diff
@@ -54,30 +63,95 @@ def initCmd (self):
 
     def setEntityFilter (self, filter):
         self.entityFilter = filter
-        print self.entityFilter
+
+    def StopRefresh(self):
+        self.timer.stop()
+        print ("Auto Refresh Stopped")
+
+    def LaunchRefresh(self):
+        print ("Auto Refreshed Launched")
+        self.timer = QtCore.QTimer()
+        self.timer.connect(self.timer, QtCore.SIGNAL("timeout()"), self.createAllGraph)
+        self.timer.start(1000)
+    
+    def UpdateFilter (self, Filt):
+        self.filter = Filt.split()
+
+    def getList (self):
+        chaine = "\n"
+        liste = dynamic_graph.entity.Entity.entityClassNameList
+        for i in liste:
+            chaine += i + "\n"
+
+        QMessageBox.information(None, "Entities",chaine,
+                        QMessageBox.Ok,
+                        QMessageBox.Ok)
+
 
     def createAllGraph (self):
-        entities = self.cmd.run ("dg.entity.Entity.entities.keys()")
+        self.blocked =""
+        self.blocked2 =""
+        again = 0
+        str_entities = self.cmd.run ("dg.entity.Entity.entities.keys()")
         self.clear()
-        for e in entities:
-            if self.entityFilter is not None and not self.entityFilter.search(e):
-                continue
-            etype = self.cmd.run("dg.entity.Entity.entities['"+e+"'].className")
-            self.types[e] = etype
-            if self.typeCallbacks.has_key(etype):
-                self.typeCallbacks[etype][0] (e)
-            else:
-                self._nodeEntity(e)
-        for e in entities:
-            if self.entityFilter is not None and not self.entityFilter.search(e):
-                continue
-            etype = self.types[e]
-            if self.typeCallbacks.has_key(etype):
-                self.typeCallbacks[etype][1] (e)
-            else:
-                self._edgeEntitySignals (e)
+        entities = eval(str_entities)
+        ent_list = entities
+        for e in ent_list:
+            for i in self.filter:
+                if i == "0" or i in e:
+                    if again != 0:
+                        pass
+                    else:
+                        if self.entityFilter is not None and not self.entityFilter.search(e):
+                            continue
+                        etype = self.cmd.run("dg.entity.Entity.entities['"+e+"'].className")
+                        self.types[e] = etype
+                        if self.typeCallbacks.has_key(etype):
+                            self.typeCallbacks[etype][0] (e)
+                        else:
+                            self._nodeEntity(e)
+                        again += 1
+            again = 0
```

Comment:
Suggestion to make the code easier to read. Note that:
- you should apply it to the loop below,
- you should test it.

```suggestion
        for e in ent_list:
            for i in self.filter:
                if i != "0" and i not in e:
                    continue
                if self.entityFilter is not None and not self.entityFilter.search(e):
                    continue
                etype = self.cmd.run("dg.entity.Entity.entities['"+e+"'].className")
                self.types[e] = etype
                if self.typeCallbacks.has_key(etype):
                    self.typeCallbacks[etype][0] (e)
                else:
                    self._nodeEntity(e)
                break
```

Reply:
Good point.
I tested your suggestion, it worked, so i used it to change the loops for the creation of entities and signals.